Forms/CrcSettingsForm.cs

```diff
@@ -15,7 +15,7 @@ public partial class CrcSettingsForm : Form
         bool configChanged = false;
         List<string> configsNeedingUnset;
         ConfigResult currentConfig;
-        Dictionary<string, dynamic> changedConfigs;
+        Dictionary<string, dynamic> changedConfigs = new Dictionary<string, dynamic>();

```

Comment:
can this be set in the constructor? or at least called by the reset function?
the idea was not to have to pre-initialize values for the form

Reply:
https://github.com/code-ready/tray-windows/blob/2c232a0c3ff2dda7cea466eb665e1dc4e4dfd63d/Forms/CrcSettingsForm.cs#L42

so this is too late? as I do not see a nullpointer exception... only when the communication does not work, but that is a postponed issue as I'd rather wanna see the issues around that now (then hide and not know about it)

Reply:
Changed to create the instance at the constructor, but don't know why it didn't work when this was created in the `getConfigurationAndResetChanged()`

but my guess is that since the exception occurs at the function `handleConfigChangesForCheckBoxes()`, and this function is called from the checkbox event handlers, it somehow needs them before the load finishes.

![vs_tray_null_exception](https://user-images.githubusercontent.com/8885742/111633698-cbd28c80-881b-11eb-8772-6897d43e2d4c.PNG)

Reply:
funny that I do not see this error... will have a closer look.
i can image that the initialize might trigger an event, but this looks like an issue that is not consistent and feels like either timing or platform-based.

Can you share the callstack from when this happens?

Reply:
Without that change I see no error:
![image](https://user-images.githubusercontent.com/1894/111636156-2c20f880-8833-11eb-96fc-606c0688c934.png)

Reply:
must be a timing issue...
The initialize happens before the load, and for some reason this times correctly for me

Reply:
Also putting the call stack here,
![call_stack_null_exception](https://user-images.githubusercontent.com/8885742/111734376-0847cc00-88a0-11eb-8694-f3cd4e5f4862.PNG)